pkg/resource/deploy/source_refresh.go

```diff
@@ -134,20 +133,29 @@ func (iter *refreshSourceIterator) newRefreshGoal(s *resource.State) (*resource.
 		if !ok {
 			return nil, errors.Errorf("unknown provider '%v' for resource '%v'", s.Provider, s.URN)
 		}
-		refreshed, err := provider.Read(s.URN, s.ID, s.Outputs)
+
+		initErrorReasons := []string{}
+		refreshed, resourceStatus, err := provider.Read(s.URN, s.ID, s.Outputs)
```

Comment:
Also for clarity, if I have a resource that was previously healthy but became unhealthy, `refresh` will populate `InitErrors` for those resources?

Reply:
Correct.